src/components/TokenRow/TokenRow.tsx

```diff
@@ -14,14 +14,40 @@ const CryptoInput = (props: InputProps) => (
   <Input
     pr='4.5rem'
     pl='7.5rem'
-    size='lg'
+    size='xl'
+    minHeight={12}
+    fontSize={getFontSizeT()}
     type='number'
     variant='filled'
     placeholder='Enter amount'
     {...props}
   />
 )
 
+var valueTrade: number | undefined
+
+const getFontSizeT = () => {
+  if (valueTrade === undefined || valueTrade === 1 || valueTrade === null) return '18px'
+  if (valueTrade !== undefined) {
+    const length = valueTrade
+    if (length >= 18) return '9px'
+    if (length >= 16) return '11px'
+    if (length >= 14) return '12px'
+    if (length >= 12) return '13px'
+    if (length >= 10) return '15px'
+    if (length >= 9) return '15px'
+    if (length >= 8) return '16px'
+    if (length >= 7) return '16px'
+    if (length >= 6) return '17px'
+    if (length >= 5) return '17px'
+    if (length >= 4) return '17px'
+    if (length >= 3) return '17px'
+    if (length >= 2) return '17px'
+    return '18px'
+  }
+  return '18px'
+}
```

Comment:
Is there not another way to do this, like using CSS? This feels pretty hacky which may not work in every browser/screen size/etc

Reply:
well, you cannot use css to check for the length of a character in a variable and reduce the size, that is my understanding  but if you feel you can sure 🙂